src/SystemCommands-MessageCommands/SycShowLocalSendersCommand.class.st

```diff
@@ -0,0 +1,81 @@
+"
+This command find the senders of a given method in the classes upper in the inheritance chain above a given method.
+"
+Class {
+	#name : 'SycShowLocalSendersCommand',
+	#superclass : 'SycMessageCommand',
+	#instVars : [
+		'selectedClass',
+		'selectedMessage'
+	],
+	#category : 'SystemCommands-MessageCommands',
+	#package : 'SystemCommands-MessageCommands'
+}
+
+{ #category : 'execution' }
+SycShowLocalSendersCommand >> applyResultInContext: a [
+]
+
+{ #category : 'accessing' }
+SycShowLocalSendersCommand >> defaultMenuIconName [
+
+	^ #edit
+]
+
+{ #category : 'accessing' }
+SycShowLocalSendersCommand >> defaultMenuItemName [
+
+	^ 'Hierarchical senders'
+]
+
+{ #category : 'execution' }
+SycShowLocalSendersCommand >> execute [
+
+	Smalltalk tools messageList browse: self searchInTheWholeHierarchy.
+]
+
+{ #category : 'testing' }
+SycShowLocalSendersCommand >> isComplexRefactoring [
+
+	^ false
+]
+
+{ #category : 'execution' }
+SycShowLocalSendersCommand >> prepareFullExecutionInContext: aToolContext [
+
+	super prepareFullExecutionInContext: aToolContext.
+	selectedMessage := aToolContext lastSelectedMessage.
```

Comment:
base class already defines messages variable retrieved from the context

Reply:
I don't understand, could you explain what you mean?

Reply:
Look into the base class SycMessageCommand>>prepareFullExecutionInContext:. It reads all selected #messages from the context (in browser you can select multiple methods and query local senders of all of them).
Thus you don't really need new inst vars in your command as all data is already there:
- #messages are instances of SycMessageDescription which has #methodClass of a method where the message is used.

I see there is a bug in how SycMessageDescription is created from the source code context. But it's easy to fix:
```Smalltalk
SycMessageDescription>>ofMessageNode: astMessageNode
	^(self fromContextOf: astMessageNode methodNode)
		selector: astMessageNode selector;
		argumentNames: (astMessageNode arguments collect: [:each | each formattedCode])
```